auth/user_mgt.go

```diff
@@ -415,17 +456,41 @@ func (u *UserToUpdate) preparePayload() (map[string]interface{}, error) {
 	processDeletion(params, "phoneNumber", "deleteProvider", "phone")
 
 	if err := processClaims(params); err != nil {
-		return nil, err
+		return err
+	}
+
+	if params["customAttributes"] != nil {
+		user.CustomAttributes = params["customAttributes"].(string)
 	}
 
 	for key, validate := range commonValidators {
 		if v, ok := params[key]; ok {
 			if err := validate(v); err != nil {
-				return nil, err
+				return err
 			}
+			reflect.ValueOf(user).Elem().FieldByName(strings.Title(key)).SetString(params[key].(string))
```

Comment:
How will this work if we had some non-string params?

Reply:
For now in commonValidators there are only string params.
And all validators in the list cast value to string

```
func validateDisplayName(val interface{}) error {
	if len(val.(string)) == 0 {
```

This is the same behavior than if we pass a boolean to password field without using the setter.

```
&UserToCreate{params: map[string]interface{}{"password": true}},
```

instead of

```
(&UserToCreate{}).Password("123456"),
```

**panic: interface conversion: interface {} is bool, not string**

@hiranya911 Any Ideas ?

Reply:
I was mostly concerned about any non-string attributes we may introduce in the future. But I think for now what you have done is ok. I'll leave it to @avishalom to make the final call.

Reply:
I'm happy to leave it like that and to add a type switch in the future if necessary.